homeassistant/components/xiaomi_miio/__init__.py

```diff
@@ -468,9 +496,27 @@ async def async_unload_entry(hass: HomeAssistant, config_entry: ConfigEntry) ->
         config_entry, platforms
     )
 
+    _LOGGER.debug("Removing subscribtions from miio device memory")
+    push_server = hass.data[DOMAIN][KEY_PUSH_SERVER]
+    device = hass.data[DOMAIN][config_entry.entry_id][KEY_DEVICE]
+    await hass.async_add_executor_job(push_server.unregister_miio_device, device)
```

Comment:
Why are we unregistering the miio device in another thread when the server seems to support asyncio?

Reply:
see comment below